<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from akros2.urdf.xacro              | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="akros2">
  <material name="FlatBlack">
    <color rgba="0.25 0.25 0.25 1"/>
  </material>
  <material name="DarkGrey">
    <color rgba="0.5 0.5 0.5 1"/>
  </material>
  <material name="Grey">
    <color rgba="0.75 0.75 0.75 1"/>
  </material>
  <material name="White">
    <color rgba="1 1 1 1"/>
  </material>
  <!-- WHEEL_LF TRANSMISSION -->
  <transmission name="trans_lf">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_lf">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_lf">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- WHEEL_LB TRANSMISSION -->
  <transmission name="trans_lb">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_lb">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_lb">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- WHEEL_RF TRANSMISSION -->
  <transmission name="trans_rf">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_rf">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_rf">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- WHEEL_RB TRANSMISSION -->
  <transmission name="trans_rb">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_rb">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_rb">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="base_link">
    <material>Gazebo/White</material>
  </gazebo>
  <gazebo reference="nav_link">
    <material>Gazebo/DarkGrey</material>
  </gazebo>
  <gazebo reference="laser_link">
    <material>Gazebo/FlatBlack</material>
  </gazebo>
  <gazebo reference="t265_pose_frame">
    <material>Gazebo/Grey</material>
  </gazebo>
  <gazebo reference="wheel_lf">
    <material>Gazebo/DarkGrey</material>
  </gazebo>
  <gazebo reference="wheel_lb">
    <material>Gazebo/DarkGrey</material>
  </gazebo>
  <gazebo reference="wheel_rf">
    <material>Gazebo/DarkGrey</material>
  </gazebo>
  <gazebo reference="wheel_rb">
    <material>Gazebo/DarkGrey</material>
  </gazebo>
  <gazebo reference="laser_frame">
    <sensor name="head_ld06_sensor" type="ray">
      <pose>0 0 0.06 0 0 0</pose>
      <visualize>false</visualize>
      <update_rate>40</update_rate>
      <ray>
        <scan>
          <horizontal>
            <samples>720</samples>
            <resolution>1</resolution>
            <min_angle>-3.14159265</min_angle>
            <max_angle>3.14159265</max_angle>
          </horizontal>
        </scan>
        <range>
          <min>0.2</min>
          <max>30.0</max>
          <resolution>0.01</resolution>
        </range>
        <noise>
          <type>gaussian</type>
          <mean>0.0</mean>
          <stddev>0.01</stddev>
        </noise>
      </ray>
      <plugin filename="libgazebo_ros_laser.so" name="gazebo_ros_head_ld06_controller">
        <topicName>scan</topicName>
        <frameName>laser_frame</frameName>
      </plugin>
    </sensor>
  </gazebo>
  <gazebo>
    <plugin filename="libgazebo_ros_imu.so" name="imu_plugin">
      <alwaysOn>true</alwaysOn>
      <bodyName>imu_frame</bodyName>
      <frameName>imu_frame</frameName>
      <topicName>imu</topicName>
      <serviceName>imu_service</serviceName>
      <gaussianNoise>0.0</gaussianNoise>
      <updateRate>200</updateRate>
      <imu>
        <noise>
          <type>gaussian</type>
          <rate>
            <mean>0.0</mean>
            <stddev>2e-4</stddev>
            <bias_mean>0.0000075</bias_mean>
            <bias_stddev>0.0000008</bias_stddev>
          </rate>
          <accel>
            <mean>0.0</mean>
            <stddev>1.7e-2</stddev>
            <bias_mean>0.1</bias_mean>
            <bias_stddev>0.001</bias_stddev>
          </accel>
        </noise>
      </imu>
    </plugin>
  </gazebo>
  <gazebo>
    <plugin filename="libgazebo_ros_planar_move.so" name="omni_steering">
      <commandTopic>cmd_vel</commandTopic>
      <odometryTopic>odom</odometryTopic>
      <odometryFrame>odom</odometryFrame>
      <odometryRate>100.0</odometryRate>
      <robotBaseFrame>base_link</robotBaseFrame>
      <publishTF>false</publishTF>
    </plugin>
  </gazebo>
  <!--<gazebo>
  <plugin name="control" filename="libgazebo_ros_control.so"/>
</gazebo>-->
  <!-- **** LINKS **** -->
  <!-- BASE_LINK -->
  <link name="base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://akros2_description/meshes/base_module.dae"/>
      </geometry>
      <material name="White"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.003 0 0.0495"/>
      <!-- If needed: offset from ground = 0.0278 -->
      <geometry>
        <box size="0.21 0.154 0.099"/>
      </geometry>
    </collision>
  </link>
  <!-- BASE_LINK_INERTIAL (WORKAROUND FOR KDL WHICH DOES NOT ACCEPT BASE_LINK WITH INERTIA) -->
  <link name="base_link_inertial">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="2.0"/>
      <inertia ixx="0.005586166666666667" ixy="0" ixz="0" iyy="0.008983499999999998" iyz="0" izz="0.008983499999999998"/>
    </inertial>
  </link>
  <!-- BASE_FOOTPRINT -->
  <link name="base_footprint"/>
  <!-- NAV_LINK -->
  <link name="nav_link">
    <inertial>
      <mass value="0.5"/>
      <inertia ixx="0.0010282083333333334" ixy="0" ixz="0" iyy="0.0017067083333333335" iyz="0" izz="0.0017067083333333335"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.099"/>
      <geometry>
        <mesh filename="package://akros2_description/meshes/navigation_module.dae"/>
      </geometry>
      <material name="DarkGrey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.0155"/>
      <!-- If needed: offset from base_link = -0.099 -->
      <geometry>
        <box size="0.2 0.154 0.031"/>
      </geometry>
    </collision>
  </link>
  <!-- LD06 LASER_LINK -->
  <link name="laser_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.12"/>
      <inertia ixx="2.5406100000000004e-05" ixy="0" ixz="0" iyy="2.533e-05" iyz="0" izz="2.533e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 -1.5707963" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://akros2_description/meshes/ld06.dae"/>
      </geometry>
      <material name="FlatBlack"/>
    </visual>
    <collision>
      <origin rpy="0 0 -1.5707963" xyz="0 0 0.0165"/>
      <geometry>
        <box size="0.038 0.0381 0.033"/>
      </geometry>
    </collision>
  </link>
  <!-- LASER_FRAME -->
  <link name="laser_frame"/>
  <!-- MPU9250 IMU_LINK -->
  <link name="imu_link"/>
  <!-- IMU_FRAME -->
  <link name="imu_frame"/>
  <!-- T265 TRACKING CAMERA (T265_POSE_FRAME AS WORKAROUND TO KEEP TREE CONTINUOUS)-->
  <link name="t265_pose_frame">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://akros2_description/meshes/t265.dae"/>
      </geometry>
      <material name="Grey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0.00125 0"/>
      <geometry>
        <box size="0.012 0.108 0.024"/>
      </geometry>
    </collision>
  </link>
  <!-- T265 TRACKING CAMERA (T265_POSE_INERTIAL)-->
  <link name="t265_pose_inertial">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.055"/>
      <inertia ixx="5.61e-05" ixy="0" ixz="0" iyy="3.3e-06" iyz="0" izz="3.3e-06"/>
    </inertial>
  </link>
  <!-- WHEEL LEFT FRONT -->
  <link name="wheel_lf">
    <inertial>
      <mass value="0.075"/>
      <inertia ixx="3.43e-05" ixy="0" ixz="0" iyy="3.43e-05" iyz="0" izz="5.4149999999999995e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://akros2_description/meshes/wheel_lf.dae"/>
      </geometry>
      <material name="FlatBlack"/>
    </visual>
    <collision>
      <origin rpy="1.5707963 0 0" xyz="0 0.017 0"/>
      <!-- If needed: offset from joint = 0.01 -->
      <geometry>
        <cylinder length="0.034" radius="0.038"/>
      </geometry>
    </collision>
  </link>
  <!-- WHEEL LEFT BACK -->
  <link name="wheel_lb">
    <inertial>
      <mass value="0.075"/>
      <inertia ixx="3.43e-05" ixy="0" ixz="0" iyy="3.43e-05" iyz="0" izz="5.4149999999999995e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://akros2_description/meshes/wheel_lb.dae"/>
      </geometry>
      <material name="FlatBlack"/>
    </visual>
    <collision>
      <origin rpy="1.5707963 0 0" xyz="0 0.017 0"/>
      <!-- If needed: offset from joint = 0.01 -->
      <geometry>
        <cylinder length="0.034" radius="0.038"/>
      </geometry>
    </collision>
  </link>
  <!-- WHEEL RIGHT FRONT -->
  <link name="wheel_rf">
    <inertial>
      <mass value="0.075"/>
      <inertia ixx="3.43e-05" ixy="0" ixz="0" iyy="3.43e-05" iyz="0" izz="5.4149999999999995e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://akros2_description/meshes/wheel_rf.dae"/>
      </geometry>
      <material name="FlatBlack"/>
    </visual>
    <collision>
      <origin rpy="1.5707963 0 0" xyz="0 -0.017 0"/>
      <!-- If needed: offset from joint = 0.01 -->
      <geometry>
        <cylinder length="0.034" radius="0.038"/>
      </geometry>
    </collision>
  </link>
  <!-- WHEEL RIGHT BACK -->
  <link name="wheel_rb">
    <inertial>
      <mass value="0.075"/>
      <inertia ixx="3.43e-05" ixy="0" ixz="0" iyy="3.43e-05" iyz="0" izz="5.4149999999999995e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://akros2_description/meshes/wheel_rb.dae"/>
      </geometry>
      <material name="FlatBlack"/>
    </visual>
    <collision>
      <origin rpy="1.5707963 0 0" xyz="0 -0.017 0"/>
      <!-- If needed: offset from joint = 0.01 -->
      <geometry>
        <cylinder length="0.034" radius="0.038"/>
      </geometry>
    </collision>
  </link>
  <!-- **** JOINTS **** -->
  <!-- BASE_LINK TO T265_POSE_FRAME -->
  <joint name="pose_base_fixed" type="fixed">
    <parent link="t265_pose_frame"/>
    <child link="base_link"/>
    <origin xyz="-0.094 0 -0.115"/>
  </joint>
  <!-- T265_POSE_INERTIAL TO T265_POSE_FRAME -->
  <joint name="pose_inertial_fixed" type="fixed">
    <parent link="t265_pose_frame"/>
    <child link="t265_pose_inertial"/>
    <origin xyz="0 0 0"/>
  </joint>
  <!-- BASE_LINK_INERTIAL TO BASE_LINK -->
  <joint name="base_inertial_fixed" type="fixed">
    <parent link="base_link"/>
    <child link="base_link_inertial"/>
    <origin xyz="0 0 0"/>
  </joint>
  <!-- BASE_FOOTPRINT TO BASE_LINK -->
  <joint name="base_footprint_fixed" type="fixed">
    <parent link="base_link"/>
    <child link="base_footprint"/>
    <origin xyz="0 0 0"/>
  </joint>
  <!-- NAV_LINK TO BASE_LINK -->
  <joint name="base_nav_fixed" type="fixed">
    <parent link="base_link"/>
    <child link="nav_link"/>
    <origin xyz="0 0 0.099"/>
    <axis xyz="0 0 0"/>
  </joint>
  <!-- LASER_LINK TO NAV_LINK -->
  <joint name="nav_laser_fixed" type="fixed">
    <parent link="nav_link"/>
    <child link="laser_link"/>
    <origin xyz="0.0632 0 0.0155"/>
  </joint>
  <!-- LASER_FRAME TO LASER_LINK -->
  <joint name="laser_frame_fixed" type="fixed">
    <parent link="laser_link"/>
    <child link="laser_frame"/>
    <origin xyz="0 0 0.0255"/>
  </joint>
  <!-- IMU_LINK TO BASE_LINK -->
  <joint name="imu_base_fixed" type="fixed">
    <parent link="base_link"/>
    <child link="imu_link"/>
    <origin xyz="0 0 0"/>
  </joint>
  <!-- IMU_FRAME TO IMU_LINK -->
  <joint name="imu_frame_fixed" type="fixed">
    <parent link="imu_link"/>
    <child link="imu_frame"/>
    <origin xyz="0 0 0"/>
  </joint>
  <!-- WHEEL_LF TO BASE_LINK -->
  <joint name="joint_lf" type="continuous">
    <parent link="base_link"/>
    <child link="wheel_lf"/>
    <origin rpy="0 0 0" xyz="0.0495 0.078 0.021"/>
    <axis xyz="0 1 0"/>
  </joint>
  <!-- WHEEL_LB TO BASE_LINK -->
  <joint name="joint_lb" type="continuous">
    <parent link="base_link"/>
    <child link="wheel_lb"/>
    <origin rpy="0 0 0" xyz="-0.0495 0.078 0.021"/>
    <axis xyz="0 1 0"/>
  </joint>
  <!-- WHEEL_RF TO BASE_LINK -->
  <joint name="joint_rf" type="continuous">
    <parent link="base_link"/>
    <child link="wheel_rf"/>
    <origin rpy="0 0 0" xyz="0.0495 -0.078 0.021"/>
    <axis xyz="0 1 0"/>
  </joint>
  <!-- WHEEL_RB TO BASE_LINK -->
  <joint name="joint_rb" type="continuous">
    <parent link="base_link"/>
    <child link="wheel_rb"/>
    <origin rpy="0 0 0" xyz="-0.0495 -0.078 0.021"/>
    <axis xyz="0 1 0"/>
  </joint>
</robot>
